src/Kis/Kis.hs

```diff
@@ -38,16 +52,39 @@ isWriteAction (CreatePatient _) = True
 isWriteAction (PlacePatient _ _) = True
 isWriteAction _ = False
 
-data Kis m =
-    Kis
-    { k_requestHandler :: forall a. KisRequest a -> m a
-      -- ^ Interface with the data
-    , k_clock :: Clock
-      -- ^ Interface with clock functions
-    }
+class KisRead m where
+    getPatients ::  m [Entity Patient]
+    getPatient :: PatientId -> m (Maybe Patient)
+
+class KisWrite m where
+    createBed :: String -> m BedId
+    createPatient :: Patient -> m PatientId
+    placePatient :: PatientId -> BedId -> m (Maybe PatientBedId)
+
+class KisClock m where
```

Comment:
Ich frage mich, ob man die Uhr wirklich von den beiden anderen Klassen so abkapseln kann oder ob nicht `KisWrite m` eigentlich auch schon Zugriff auf die Uhr braucht, weil z.B. die virtuelle Zeit bei `createBed` etc. mit in die Datenbank geschrieben werden müsste.
Auf der anderen Seite wäre zusätzlich eine `KisNotification`-Typklasse denkbar, welche die Notifications in die Datenbank schreibt und wieder ausliest. Können wir vielleicht aber erst mal defern. Ich weiß jedenfalls noch nicht, ob und was es bringt und wie die Funktionalität aussehen müsste.